Fixed issue #1

aaroncohen73 opened the pull request:
@matthewrsj @Kennric Could you please test this fix?

matthewrsj commented on climesync.py:
```diff
@@ -71,12 +73,11 @@
-        pp.pprint(ts.authenticate(username, password, auth_type))
+        username = arg_username if arg_username else raw_input("username: ")
+
+        password = arg_password if arg_password else raw_input("password: ")
```
What happens if you sign in like so:
```
./climesync.py -c <baseurl> -u user1 -p pass1
```
but then you want to change your user (think admin switching to his own account)
```
>> s
hey! you are already signed in because arg_username and arg_password are set.  
Here is a brand new token for the same user you set in the command line args!
```

matthewrsj commented:
Tested, and this appears to fix the bug. Good job!

A couple notes:

- Try to be a little more descriptive in your commit titles in the future. `fixed issue ##` is not very helpful when looking at history. A more descriptive example could be `allow re-auth after setting command-line args`. Then go more in-depth in the actual commit method below.

- Now that we have this in an OSL repository, let's add comments to the code we modify. I know I didn't do this before when it was my own project, but we are following a development workflow now.

Once you add some comments to your code to explain what is going on, I'll give a +1.

matthewrsj commented:
Oops, just found a bug in your bugfix:

```
mrsj@pewter:~/work/climesync [cohenaa/login-fix] (venv) » climesync.py --connect http://timesync-staging.osuosl.org/v1 --user test
===============================================================
 pymesync CLI to interact with TimeSync
===============================================================

What do you want to do?
c - connect
s - sign in

ct - submit time
ut - update time
gt - get times

cp - create project
up - update project
gp - get projects

ca - create activity
ua - update activity
ga - get activities

cu - create user
uu - update user
gu - get users

m - print this menu
q - exit

>> s
password: test
{   u'token': u'eyJ0eXAiOiJKV1QiLCJhbGciOiJITUFDLVNIQTUxMiJ9.eyJpc3MiOiJvc3Vvc2wtdGltZXN5bmMtc3RhZ2luZyIsInN1YiI6InRlc3QiLCJleHAiOjE0NTk3ODUxMDk1NzYsImlhdCI6MTQ1OTc4MzMwOTU3Nn0=.Zxl23MvvHt/eymDP46vQlS0Z+RHuawbl4pnzGn82/nvlkH46FweHG8X1Jo+Nk6FSAcSGdDvFvDCJ1ZWqqLjZyg=='}
>> s
password: test
{   u'token': u'eyJ0eXAiOiJKV1QiLCJhbGciOiJITUFDLVNIQTUxMiJ9.eyJpc3MiOiJvc3Vvc2wtdGltZXN5bmMtc3RhZ2luZyIsInN1YiI6InRlc3QiLCJleHAiOjE0NTk3ODUxMTM1NDMsImlhdCI6MTQ1OTc4MzMxMzU0M30=.IiYqV2Sc1Lc+2+AV2/FXpMfNgGSJeCDzJYZgyvv18ygqlIW4mSK2krfjoQkB3s7hlj4TGdvh0b0M6+6gIEpurQ=='}
>> 
```

can't sign in as a different user if *just one* sign-in flag is set on the command line.

aaroncohen73 commented:
@matthewrsj That's the way that I interpreted the design. i.e. you specify the server and user as command line arguments and then enter your password inside the program. If that isn't the intended functionality, it's easy enough to change it.

aaroncohen73 commented:
@matthewrsj @Kennric I've fixed the issue with being unable to sign in as a different use by adding a sign out command and I've also added more comments to my code. Could you review the new commit?

matthewrsj commented:
@aaroncohen73 please add the new `so` option to the README. Once that is done, +1.

aaroncohen73 commented:
@matthewrsj Done

matthewrsj commented:
+1, I'll be merging your PRs when I +1 them so we can get this stable ASAP. Don't worry about getting more than one +1 for climesync.
